File: ios/Sasquatsh/Views/Raffle/RaffleBannerView.swift
import SwiftUI

struct RaffleBannerView: View {
    let raffle: Raffle

    var body: some View {
        VStack(spacing: 0) {
            ZStack(alignment: .topTrailing) {
                // Banner image or gradient
                if let url = resolveImageURL(raffle.bannerImageUrl) ?? resolveImageURL(raffle.prizeImageUrl) {
                    AsyncImage(url: url) { image in
                        image.resizable().aspectRatio(contentMode: .fill)
                    } placeholder: {
                        bannerGradient
                    }
                    .frame(height: 120)
                    .clipped()
                } else {
                    bannerGradient
                        .frame(height: 120)
                }

                // Monthly Raffle label overlay
                Text("RAFFLE")
                    .font(.system(size: 10, weight: .bold))
                    .foregroundStyle(.white)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 4)
                    .background(.black.opacity(0.4))
                    .clipShape(RoundedRectangle(cornerRadius: MD3Shape.small))
                    .padding(8)
            }

            // Info section
            VStack(alignment: .leading, spacing: 8) {
                HStack {
                    Text("Monthly Raffle")
                        .font(.md3LabelSmall)
                        .foregroundStyle(Color.md3Primary)
                    Spacer()
                    if let value = raffle.prizeValueFormatted {
                        Text(value)
                            .font(.md3LabelLarge)
                            .foregroundStyle(Color.md3Tertiary)
                    }
                }

                Text(raffle.prizeName)
                    .font(.md3TitleMedium)
                    .foregroundStyle(Color.md3OnSurface)

                if let desc = raffle.description, !desc.isEmpty {
                    Text(desc)
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3OnSurfaceVariant)
                        .lineLimit(2)
                }

                Divider()

                // Stats row
                HStack(spacing: 16) {
                    if let time = raffle.timeRemaining {
                        Label(time, systemImage: "clock")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }

                    if let stats = raffle.stats {
                        Label("\(stats.displayTotalEntries) entries", systemImage: "ticket")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)

                        Label("\(stats.displayParticipants) players", systemImage: "person.2")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }
                }

                // User's entries
                if let total = raffle.userTotalEntries, total > 0 {
                    HStack(spacing: 6) {
                        Image(systemName: "star.fill")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3Primary)
                        Text("You have \(total) entr\(total == 1 ? "y" : "ies")")
                            .font(.md3BodySmall)
                            .fontWeight(.medium)
                            .foregroundStyle(Color.md3Primary)
                    }
                } else {
                    HStack(spacing: 6) {
                        Image(systemName: "info.circle")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                        Text("Host or attend games to earn entries!")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }
                }
            }
            .padding(12)
        }
        .background(Color.md3Surface)
        .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
        .overlay(
            RoundedRectangle(cornerRadius: MD3Shape.medium)
                .stroke(Color.md3OutlineVariant, lineWidth: 1)
        )
    }

    private func resolveImageURL(_ urlStr: String?) -> URL? {
        guard let urlStr, !urlStr.isEmpty else { return nil }
        if urlStr.hasPrefix("http") {
            return URL(string: urlStr)
        }
        // Relative path — resolve against website
        return URL(string: "https://sasquatsh.com\(urlStr)")
    }

    private var bannerGradient: some View {
        LinearGradient(
            colors: [.orange, .red.opacity(0.8)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
        .overlay(
            VStack {
                Image(systemName: "trophy.fill")
                    .font(.system(size: 36))
                    .foregroundStyle(.white.opacity(0.6))
            }
        )
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/raffle/RaffleBannerView.kt
package com.sasquatsh.app.views.raffle

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.ConfirmationNumber
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Group
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.sasquatsh.app.models.Raffle

@Composable
fun RaffleBannerView(
    raffle: Raffle,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        shape = RoundedCornerShape(12.dp)
    ) {
        Column {
            // Banner image or gradient
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(120.dp)
            ) {
                val imageUrl = resolveImageUrl(raffle.bannerImageUrl)
                    ?: resolveImageUrl(raffle.prizeImageUrl)

                if (imageUrl != null) {
                    AsyncImage(
                        model = imageUrl,
                        contentDescription = raffle.prizeName,
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(120.dp),
                        contentScale = ContentScale.Crop
                    )
                } else {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(120.dp)
                            .background(
                                brush = Brush.linearGradient(
                                    colors = listOf(Color(0xFFFFA500), Color(0xFFCC3333))
                                )
                            ),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            Icons.Default.EmojiEvents,
                            contentDescription = null,
                            modifier = Modifier.size(36.dp),
                            tint = Color.White.copy(alpha = 0.6f)
                        )
                    }
                }

                // RAFFLE label
                Text(
                    "RAFFLE",
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier
                        .align(Alignment.TopEnd)
                        .padding(8.dp)
                        .background(
                            Color.Black.copy(alpha = 0.4f),
                            RoundedCornerShape(4.dp)
                        )
                        .padding(horizontal = 8.dp, vertical = 4.dp)
                )
            }

            // Info section
            Column(modifier = Modifier.padding(12.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(
                        "Monthly Raffle",
                        style = MaterialTheme.typography.labelSmall,
                        color = MaterialTheme.colorScheme.primary
                    )
                    raffle.prizeValueFormatted?.let { value ->
                        Text(
                            value,
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.tertiary
                        )
                    }
                }

                Text(
                    raffle.prizeName,
                    style = MaterialTheme.typography.titleMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )

                raffle.description?.let { desc ->
                    if (desc.isNotEmpty()) {
                        Text(
                            desc,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            maxLines = 2,
                            overflow = TextOverflow.Ellipsis
                        )
                    }
                }

                Spacer(modifier = Modifier.height(8.dp))
                HorizontalDivider()
                Spacer(modifier = Modifier.height(8.dp))

                // Stats row
                Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                    raffle.stats?.let { stats ->
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.ConfirmationNumber,
                                contentDescription = null,
                                modifier = Modifier.size(14.dp),
                                tint = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                "${stats.displayTotalEntries} entries",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.Group,
                                contentDescription = null,
                                modifier = Modifier.size(14.dp),
                                tint = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                "${stats.displayParticipants} players",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }

                Spacer(modifier = Modifier.height(4.dp))

                // User entries
                val total = raffle.userTotalEntries ?: 0
                if (total > 0) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            Icons.Default.Star,
                            contentDescription = null,
                            modifier = Modifier.size(14.dp),
                            tint = MaterialTheme.colorScheme.primary
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            "You have $total entr${if (total == 1) "y" else "ies"}",
                            style = MaterialTheme.typography.bodySmall,
                            fontWeight = FontWeight.Medium,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                } else {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            Icons.Default.Info,
                            contentDescription = null,
                            modifier = Modifier.size(14.dp),
                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            "Host or attend games to earn entries!",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }
            }
        }
    }
}

private fun resolveImageUrl(urlStr: String?): String? {
    if (urlStr.isNullOrEmpty()) return null
    return if (urlStr.startsWith("http")) urlStr
    else "https://sasquatsh.com$urlStr"
}
